ChattoAdditions/Source/Chat Items/CompoundMessage/CompoundMessagePresenter.swift

```diff
@@ -67,60 +70,96 @@ public final class CompoundMessagePresenter<ViewModelBuilderT, InteractionHandle
         )
     }
 
-    public override var canCalculateHeightInBackground: Bool {
+    open override var canCalculateHeightInBackground: Bool {
         return true
     }
 
-    public override class func registerCells(_ collectionView: UICollectionView) {
-        collectionView.register(CompoundMessageCollectionViewCell<ModelT>.self,
-                                forCellWithReuseIdentifier: .compoundCellReuseId)
+    open override class func registerCells(_ collectionView: UICollectionView) {
+        // Cell registration is happening lazily, right before the moment when a cell is dequeued.
     }
 
-    public override func dequeueCell(collectionView: UICollectionView, indexPath: IndexPath) -> UICollectionViewCell {
-        return collectionView.dequeueReusableCell(withReuseIdentifier: .compoundCellReuseId,
-                                                  for: indexPath)
+    open override func dequeueCell(collectionView: UICollectionView, indexPath: IndexPath) -> UICollectionViewCell {
+        let cellReuseIdentifier = self.compoundCellReuseId
+        collectionView.register(CompoundMessageCollectionViewCell.self, forCellWithReuseIdentifier: cellReuseIdentifier)
+        return collectionView.dequeueReusableCell(withReuseIdentifier: cellReuseIdentifier, for: indexPath)
     }
 
-    public override func heightForCell(maximumWidth width: CGFloat,
+    open override func heightForCell(maximumWidth width: CGFloat,
```

Comment:
Indentations are not correct for some functions now after changing access modifier from `public` to `open`

Reply:
Fixed.